src/components/countryDetails/CountryDetails.tsx

```diff
@@ -1,45 +1,214 @@
+										<StyledCountryName>{name.common}</StyledCountryName>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Native Name:
+											</StyledSpanWithCategoryName>{" "}
+											{altSpellings[2]}
+										</StyledParagraphWithCountryInfo>
+
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Population:
+											</StyledSpanWithCategoryName>{" "}
+											{population}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Region:
+											</StyledSpanWithCategoryName>{" "}
+											{region}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Sub Region:
+											</StyledSpanWithCategoryName>{" "}
+											{subregion}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Capital:
+											</StyledSpanWithCategoryName>{" "}
+											{capital}
+										</StyledParagraphWithCountryInfo>
+									</StyledLeftColumnWithCountryInfo>
+									<div>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Top Level Domain:
+											</StyledSpanWithCategoryName>{" "}
+											{tld[0]}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Currencies:
+											</StyledSpanWithCategoryName>{" "}
+											{currencieName}
+										</StyledParagraphWithCountryInfo>
+										<StyledParagraphWithCountryInfo>
+											<StyledSpanWithCategoryName>
+												Languages:
+											</StyledSpanWithCategoryName>{" "}
+											{languagesKeys.map((lang) => {
+												return (
+													<span key={lang}>
+														{languages[lang as unknown as number]},{" "}
+													</span>
+												);
+											})}
+										</StyledParagraphWithCountryInfo>
+									</div>
+								</StyledDivWithCountryInfo>
+								<StyledDivWithButtons>
+									<StyledSpanWithCategoryName>
+										Border Countries:
+									</StyledSpanWithCategoryName>{" "}
+									{arr.length
+										? arr.map((country, index) => {
+												return (
+													<StyledLinkWithBorderCountries
+														to={`/${country?.name.common}`}
+														key={index}
+														type="button"
+													>
+														{country?.name.common.substr(0, 12)}
+													</StyledLinkWithBorderCountries>
+												);
+										  })
+										: "No border countries found."}
+								</StyledDivWithButtons>
+							</div>
+						);
+					}
+				)}
+			</StyledDivWithCountryData>
+		</StyledSection>
 	);
 };
 
 export const countryDetailsLoader: LoaderFunction = async ({ params }) => {
```

Comment:
Please move this api call to some proper place, like api folder. I don;t think you have to declare this function here. Please do some research 👍

Reply:
fixed.